src/mini-cast-bar/index.js

```diff
@@ -0,0 +1,154 @@
+
+import React from "react";
+import { View, Image, TouchableOpacity, Text , Appearance} from "react-native";
+import { useMiniPlayerSessionListenerHook,useMiniPlayerMediaStatusListenerHook,useMiniPlayerStyleHook,useMiniPlayerSessionInfoHook, useMiniPlayerActionsHook } from "./hooks";
+import { getTitleFromVideoInfo,getSubTitleFromVideoInfo, getApperanceColor } from "./helpers";
+import PropTypes from "prop-types";
+import { Images } from "./constants";
+
+const VizbeeMiniCastBar = ({
+  thumbnailImageSourceDefault,
+  backgroundLayerStyle,
+  progressViewStyle,
+  titleStyle,
+  subTitleStyle,
+  buttonStyle,
+  disableButtonStyle,
+  playActionButtonType,
+  shouldDisablePlayPauseButtonForLive,
+  buttonImgPlay,
+  buttonImgPause,
+  buttonImgStop
+}) => {
+
+  const isDark = Appearance.getColorScheme() === "dark";
+
+  const defaultApperance = getApperanceColor(isDark);
+
+  const {getDefaultStyle} = useMiniPlayerStyleHook({
+    backgroundLayerStyle,
+    progressViewStyle,
+    titleStyle,
+    subTitleStyle,
+    buttonStyle,
+    disableButtonStyle,
+    defaultApperance
+    });
+  
+  const defaultStyles = getDefaultStyle()
+
+  const {isVisible,setIsVisible} = useMiniPlayerSessionListenerHook();
+
+  const {
+    mediaStatus, 
+    isButtonDisabled, 
+    buttonImg, 
+    buttonType} = useMiniPlayerMediaStatusListenerHook({
+      isVisible,
+      setIsVisible,
+    playActionButtonType,
+    shouldDisablePlayPauseButtonForLive,
+    buttonImgPlay,
+    buttonImgPause,
+    buttonImgStop
+    });
+
+    const { castingTo } = useMiniPlayerSessionInfoHook({isVisible})
+
+    const {onMiniPlayerViewPress,onButtonPress} = useMiniPlayerActionsHook()
+
+  const {title, subTitle, imageURL, streamPosition, streamDuration, isLive } = mediaStatus;
+
+  // Calculte the streamPosition percentage for progress view
+  const percentage = (streamPosition / streamDuration) * 100;
+
+
+  return isVisible ? (
```

Comment:
Can we move this functional component to a .js file something like VZBMiniCastBarComponent.js as It's not a common best practice to write a functional component directly in the index.js file of a React application?
```
src
|-- components
|   |-- mini-cast-bar
|       |-- hooks
|          |-- index.js
|          |-- useMiniPlayerStyleHook
|          |-- ...
|       |-- index.js
|       |-- VZBMiniCastBarComponent.js
```

Reply:
Done